src/StepAdaptationController/src/StepAdaptationController.cpp

```diff
@@ -0,0 +1,541 @@
+
+// std
+#define NOMINMAX
+#include <algorithm>
+
+// yarp
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Value.h>
+
+// iDynTree
+#include <iDynTree/Core/EigenSparseHelpers.h>
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/Core/SpatialAcc.h>
+#include <iDynTree/Core/Direction.h>
+
+#include <WalkingControllers/StepAdaptationController/StepAdaptationController.hpp>
+#include <WalkingControllers/YarpUtilities/Helper.h>
+#include <WalkingControllers/iDynTreeUtilities/Helper.h>
+
+typedef Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> MatrixXd;
+using namespace WalkingControllers;
+
+StepAdaptationController::StepAdaptationController(const int &inputSize, const int &numberOfAllConstraints)
+    : m_inputSize(inputSize)
+    , m_numberOfConstraints(numberOfAllConstraints)
+    , m_xPositionsBuffer(2)
+    , m_yPositionsBuffer(2)
+    , m_zPositionsBuffer(3)
+    , m_zzPositionsBuffer(2)
+    , m_yawsBuffer(2)
+    , m_timesBuffer(2)
+    , m_zTimesBuffer(3)
+    , m_zzTimesBuffer(2)
+{
+    // instantiate the solver class
+    m_QPSolver = std::make_unique<OsqpEigen::Solver>();
+
+    //set the number of deceision variables of QP problem
+    m_QPSolver->data()->setNumberOfVariables(inputSize);
+
+    // set the number of all constraints includes inequality and equality constraints
+    m_QPSolver->data()->setNumberOfConstraints(numberOfAllConstraints);
+
+    m_QPSolver->settings()->setVerbosity(false);
+    m_QPSolver->settings()->setPolish(true);
```

Comment:
If you are using only ``osqp`` remove ``qpOases``.

Reply:
For the time being we are using qpOases because osqp had a problem (for the exact similar matrix example qpOases gave us a correct answer), when we were struggling to submit the humanoids paper.

Reply:
I would prefer one of these two cases:
- keep both the solvers and have a configuration parameter to decide which one to use
- have a single solver (you decide which one)

Reply:
Personally, I'm happy with `qpOases` because it's currently working nice.